python/foglamp/services/core/service_registry/service_registry.py

```diff
@@ -103,6 +103,7 @@ def unregister(cls, service_id):
         """
         expunged_service = cls._expunge(service_id, ServiceRecord.Status.Down)
         cls._logger.info("Stopped {}".format(str(expunged_service)))
+        cls.remove_from_registry(service_id)
```

Comment:
May be, We should move this as first action to delete service method, if service's status is DOWN then remove from service registry.